### DISCUSSION TITLE
i taked this message while publish animate to .aab

### DISCUSSION THREAD

**Discussion by orcuncun** [airsdk/Adobe-Runtime-Support#1184] [Q&A]
![242720109_10160014897702164_6421331699446610072_n](https://user-images.githubusercontent.com/29103010/134495508-b98d453c-47df-4ec5-a73c-9f5d5955e141.jpg)

**Comment by ajwfrost** [comment]
The error here is "Installed Build Tools revision 31.0.0.is corrupted" - and actually, it's not corrupted, it's just missing the dx.jar file that the Gradle plug-in is expecting.

If you have another Build Tools folder, try copying that across e.g.
`Android\Sdk\build-tools\30.0.2\lib\dx.jar` to `Android\Sdk\build-tools\31.0.0\lib\dx.jar`
and
`Android\Sdk\build-tools\30.0.2\dx.bat` to `Android\Sdk\build-tools\31.0.0\dx.bat`

The proper solution is actually to update to use the latest gradle plug-in and Android Studio versions so that it knows it doesn't need the DX component so we'll try to address this in a future SDK update...

**Reply by joypalsingh** [reply]
I am getting this error after adding dx jar/bat files  (AIR 33.1.1.633)
![image](https://user-images.githubusercontent.com/22320065/135709742-3ebdd28c-2a80-45f3-b83a-efcc994b4eeb.png)

**Reply by marchbold** [reply]
That looks like you may still be using distriqt's multidex ane? Try removing it as I believe AIR now includes this by default.

Cheers

**Comment by orcuncun** [comment]
i have one Build Tools folder 31.0.0
what should I do?

**Reply by ajwfrost** [reply]
Below is the required dx bat/jar file, please unzip this into the build tools folder

[dx.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/7217222/dx.zip)

**Reply by orcuncun** [reply]
am i copying to this address?
 C:\Users\tezca\AppData\Local\Android\Sdk\build-tools\31.0.0\lib

**Reply by ajwfrost** [reply]
If you extract that into the 31.0.0 folder, it puts the .bat file there and the .jar file under "lib"....
thanks

**Reply by Arkonskirrid** [reply]
Please help me, I haven't been able to sleep for a week and I get various errors
---------------------------
Adobe Animate
---------------------------
Ошибка создания файлов.

Неизвестная ошибка.
unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01

Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01

Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01

Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01



FAILURE: Build failed with an exception.



* What went wrong:

Execution failed for task ':app:mergeReleaseResources'.

> Multiple task action failures occurred:

   > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade

      > AAPT2 aapt2-4.1.1-6503028-windows Daemon #0: Unexpected error during compile 'C:\Users\Arkon\AppData\Local\Temp\3235c531-da8f-4545-8c84-e3f920785917\app\build
---------------------------
ОК   
---------------------------

**Reply by ajwfrost** [reply]
That's a new one, I've never seen that before:
 > AAPT2 aapt2-4.1.1-6503028-windows Daemon #0: Unexpected error during compile 'C:\Users\Arkon\AppData\Local\Temp\3235c531-da8f-4545-8c84-e3f920785917\app\build

My suggestion would be if you fine the AIR SDK's "lib" folder and edit the adt.cfg file, there should be a setting "KeepAndroidStudioOutput" that you can uncomment to set it to 'true'. If you run it again after that, you should find a folder "AndroidStudioProject" is created where the .aab file should have been output: this folder can then be opened within Android Studio.

This would then allow you to try out something such as downloading the 30.0.0 build-tools and trying to re-build using those..

In the meantime, to help us look into this further, if you can create the AndroidStudioProject folder and send it zipped up over to us to examine, we can try to see what's going on with this resource merging step...

thanks

**Comment by paulp-games** [comment]
Hello! Please help, got this error
Adobe Animate 23.0.1
AIR 33.1.1.935
![Screenshot 2023-05-03 125551](https://user-images.githubusercontent.com/56564741/235886035-67bed3a6-389a-4094-9d1c-0426e45f0b96.png)

**Reply by paulp-games** [reply]
![android](https://user-images.githubusercontent.com/56564741/235888408-4be44719-3a9f-4654-b79a-d95d71793b24.png)
![cfg](https://user-images.githubusercontent.com/56564741/235888630-7e701479-e248-47a9-94d7-c0655a268e91.png)

**Reply by ajwfrost** [reply]
@paulp-games the error message says you need to explicitly set the 'exported' property for your activity, because there is an intent filter.. so, I would assume within your application descriptor file, you have an additional activity - or maybe this is coming in from an ANE? The first step there would be to ensure you're using the latest versions of the ANEs...

**Reply by paulp-games** [reply]
@ajwfrost Thanks!
Got the new error
![aab](https://user-images.githubusercontent.com/56564741/235894634-6c7e6407-72e0-4ce1-9de8-3921111bfd06.png)

**Reply by ajwfrost** [reply]
Okay cool .. so that's just a warning, you don't actually need to sign AAB files. The ADT logging should tell you exactly why it's giving that message, in case you wanted to fix it..

thanks
